worlds/lingo/__init__.py

```diff
@@ -0,0 +1,149 @@
+                if data.code is not None and len(data.painting_ids) > 0
+            },
+            'panel_ids_by_location_id': {
+                data.code: data.panel_ids() for data in self.static_locat.ALL_LOCATION_TABLE.values()
+                if data.code is not None
+            },
+            'seed': self.multiworld.per_slot_randoms[self.player].randint(0, 1000000),
+        }
+
+    def generate_early(self):
+        self.player_logic = LingoPlayerLogic(self.multiworld, self.player, self.static_logic, self.test_options)
+
+    def create_regions(self):
+        create_regions(self.multiworld, self.player, self.static_logic, self.player_logic)
+
+    def create_items(self):
+        pool = []
+
+        for name in self.player_logic.REAL_ITEMS:
+            new_item = self.create_item(name)
+            pool.append(new_item)
+
+        for location, item in self.player_logic.EVENT_LOC_TO_ITEM.items():
+            event_item = LingoItem(item, ItemClassification.progression, None, player=self.player)
+            location_obj = self.multiworld.get_location(location, self.player)
+            location_obj.place_locked_item(event_item)
+
+        if self.player_logic.FORCED_GOOD_ITEM != "":
+            new_item = self.create_item(self.player_logic.FORCED_GOOD_ITEM)
+            location_obj = self.multiworld.get_location("Starting Room - HI", self.player)
+            location_obj.place_locked_item(new_item)
+
+        item_difference = len(self.player_logic.REAL_LOCATIONS) - len(pool)
+        if item_difference > 0:
+            trap_percentage = get_option_value(self.multiworld, self.player, "trap_percentage")
+            traps = floor(item_difference * trap_percentage / 100.0)
+            non_traps = item_difference - traps
+
+            for i in range(0, non_traps):
+                pool.append(self.create_item("Nothing"))
+
+            if traps > 0:
+                traps_list = ["Slowness Trap", "Iceland Trap", "Atbash Trap"]
+
+                for i in range(0, traps):
+                    pool.append(self.create_item(traps_list[i % len(traps_list)]))
+
+        self.multiworld.itempool += pool
+
+    def create_item(self, name: str) -> Item:
+        item = StaticLingoItems.ALL_ITEM_TABLE[name]
+
+        if item.progression:
+            classification = ItemClassification.progression
+        elif item.trap:
+            classification = ItemClassification.trap
+        else:
+            classification = ItemClassification.filler
+
+        new_item = LingoItem(name, classification, item.code, player=self.player)
+        return new_item
+
+    def set_rules(self):
+        set_rules(self.multiworld, self.player, self.player_logic)
+
+        self.multiworld.completion_condition[self.player] = \
+            lambda state: state.has("Victory", self.player)
+
+    def fill_slot_data(self):
+        slot_data = self._get_slot_data()
+
+        for option_name in lingo_options:
```

Comment:
This is generally bad practice. You should exclude unneeded options from this instead of just passing all of them.